### PR TITLE
OAK-11875 : avoid removing props/docs if they have empty split props

### PR DESCRIPTION
**Pull request by rishabhdaim**
@shodaaan could you please review ?

### CODE REVIEW

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java [hunk 1]
```diff
@@ -1477,6 +1478,24 @@ private void collectDeletedProperties(final NodeDocument doc, final GCPhases pha
...
+                final Set<String> splitProps = doc.getSplitPropertyNames();
+                // Skip optimization if there are no split properties
+                if (!splitProps.isEmpty()) {
+                    // Only calculate difference if we have split properties
+                    Set<String> propsToBeDeleted = SetUtils.difference(properties, retainPropSet);
+
+                    // Check for intersection between sets directly
+                    if (!Collections.disjoint(splitProps, propsToBeDeleted)) {
+                        if (AUDIT_LOG.isInfoEnabled()) {
+                            AUDIT_LOG.info("<Skipping> empty props deletion in [{}] due to presence of deleted Split Properties.", doc.getId());
```
Comment (shodaaan) [inline oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java]: Suggestion: include the intersection between splitProps and propsToBeDeleted (the actual properties which prevent deletion) in this log message. It also may be useful to include the propsToBeDeleted as well in the log message

File: oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/NodeDocumentTest.java [hunk 2]
```diff
@@ -1309,6 +1309,75 @@ public <T extends Document> T find(Collection<T> collection,
         ns.dispose();
     }
 
+    @Test
```
Comment (shodaaan) [inline oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/NodeDocumentTest.java:1312]: Suggestion: add 1 more unit test for the case when deletion of empty props is prevented by props existing in split documents.

**Review by shodaaan — COMMENTED** [review]
Looks good, added suggestions - I think logging the props that were prevented from deletion is quite important.

File: oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/NodeDocumentTest.java (hunk 2 above)
Comment (rishabhdaim) [reply oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/NodeDocumentTest.java:1312]: This has already been added in `VGCIT` class

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java (hunk 1 above)
Comment (rishabhdaim) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java]: added in https://github.com/apache/jackrabbit-oak/pull/2489/commits/fb66913c6d82bcea7e93ed810e3f4707a189488d

**Review by shodaaan — APPROVED** [review]
Looks good to me.

**Review by Joscorbe — APPROVED** [review]
